lib/daru/io/importers/csv.rb

```diff
@@ -3,14 +3,37 @@
 module Daru
   module IO
     module Importers
-      # CSV Importer Class, that extends `from_csv` method to `Daru::DataFrame`
+      # CSV Importer Class, that extends `read_csv` method to `Daru::DataFrame`
       class CSV < Base
-        Daru::DataFrame.register_io_module :from_csv, self
+        Daru::DataFrame.register_io_module :read_csv, self
 
-        # Imports a `Daru::DataFrame` from a .csv / .csv.gz file.
+        # Checks for required gem dependencies of CSV Importer
+        def initialize
+          require 'csv'
+          require 'open-uri'
+          require 'zlib'
+        end
+
+        # Reads data from a csv / csv.gz file
+        #
+        # @param path [String] Path to csv / csv.gz file, where the dataframe is to be imported
+        #   from.
+        #
+        # @return [Daru::IO::Importers::CSV]
+        #
+        # @example Reading from csv file
+        #   instance = Daru::IO::Importers::CSV.read("matrix_test.csv")
+        #
+        # @example Reading from csv.gz file
+        #   instance = Daru::IO::Importers::CSV.read("matrix_test.csv.gz")
+        def read(path)
```

Comment:
I'm having a dilemma about YARD documenting the `read` (and `from`) method. Basically, there are 2 `read` methods - `Importers::Format#read` class instance method and `Importers::Base#read` class method.

So, I'm showcasing example of `Daru::IO::Importers::CSV.read(path)` usage though it's technically the `Daru::IO::Importers::CSV.new.read(path)` which is being called by `Daru::IO::Importers.read(path)`. So, though it's technically incorrect, I'd like it to stay this way as users would find the method's showcase easier this way.

Reply:
You can imitate docs for `Daru::IO::Importers::CSV.read` (as if it exists and defined in this file) with YARD's [@!method](http://www.rubydoc.info/gems/yard/file/docs/Tags.md#method) directive. I believe it would be ideal solution from all possible points of view, WDYT?